### PR TITLE
fix: Fix CI running analysis-stats incorrectly against the standard libraries

### PR DESCRIPTION
**Pull request by Veykril**
Fixes https://github.com/rust-lang/rust-analyzer/issues/18326

### CODE REVIEW

**Comment by ShoyuVanilla** [comment]
```
thread 'main' panicked at crates/hir-ty/src/infer/cast.rs:423:13:
internal error: entered unreachable code: `[Int(I8); Bytes([32, 0, 0, 0, 0, 0, 0, 0], Empty)]` should be sized but is not?
```
It seems that we are not getting `Sized` trait from lang items 🤔

**Comment by Veykril** [comment]
Yes, we set the test cfg which causes core to be empty hence no lang items

**Comment by Veykril** [comment]
Ugh, now its stuck in const eval somewhere

**Comment by Veykril** [comment]
Okay now it should work
@bors r+

File: crates/rust-analyzer/src/cli/analysis_stats.rs [hunk 1]
```diff
@@ -59,13 +60,17 @@ impl flags::AnalysisStats {
             Rand32::new(seed)
         };
 
+        let mut cfg_overrides = CfgOverrides::default();
+        cfg_overrides.global =
+            CfgDiff::new(vec![CfgAtom::Flag(hir::sym::miri.clone())], vec![]).unwrap();
```
Comment (Veykril) [inline crates/rust-analyzer/src/cli/analysis_stats.rs:65]: This probably should be a default for the cli commands

**Comment by Veykril** [comment]
Clippy always has the last word
@bors r+

**Pull request merged**